datasets/wili_2018/wili_2018.py

```diff
@@ -0,0 +1,334 @@
+    "zea",
+    "pol",
+    "est",
+    "scn",
+    "vls",
+    "stq",
+    "gag",
+    "grn",
+    "kaz",
+    "ben",
+    "pcd",
+    "bjn",
+    "krc",
+    "amh",
+    "diq",
+    "ltz",
+    "ita",
+    "kab",
+    "bel",
+    "ang",
+    "mhr",
+    "che",
+    "koi",
+    "glv",
+    "ido",
+    "fao",
+    "bak",
+    "isl",
+    "bcl",
+    "tet",
+    "jpn",
+    "kur",
+    "map-bms",
+    "tyv",
+    "olo",
+    "arg",
+    "ori",
+    "lim",
+    "tel",
+    "lin",
+    "roh",
+    "sqi",
+    "xho",
+    "mlg",
+    "fas",
+    "hbs",
+    "tam",
+    "aze",
+    "lad",
+    "nob",
+    "sin",
+    "gla",
+    "nap",
+    "snd",
+    "ast",
+    "mal",
+    "mdf",
+    "tsn",
+    "nds",
+    "tgl",
+    "nno",
+    "sun",
+    "lzh",
+    "jbo",
+    "crh",
+    "pap",
+    "oci",
+    "hak",
+    "uzb",
+    "zho",
+    "hsb",
+    "sme",
+    "mlt",
+    "vep",
+    "lez",
+    "nld",
+    "nds-nl",
+    "mrj",
+    "spa",
+    "ceb",
+    "ina",
+    "heb",
+    "hun",
+    "que",
+    "kaa",
+    "mar",
+    "vec",
+    "frp",
+    "ell",
+    "sah",
+    "eus",
+    "ces",
+    "slk",
+    "chr",
+    "lij",
+    "nep",
+    "srd",
+    "ilo",
+    "be-tarask",
+    "bod",
+    "orm",
+    "war",
+    "glg",
+    "mon",
+    "gle",
+    "min",
+    "ibo",
+    "ile",
+    "epo",
+    "lav",
+    "lrc",
+    "als",
+    "mzn",
+    "rup",
+    "fur",
+    "tat",
+    "myv",
+    "pan",
+    "ton",
+    "kom",
+    "wuu",
+    "tcy",
+    "tuk",
+    "kan",
+    "ltg",
+]
+
+
+class Wili2018(datasets.GeneratorBasedBuilder):
+    """WiLI Language Identification Dataset"""
+
+    VERSION = datasets.Version("1.1.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="WiLI-2018 dataset",
+            version=VERSION,
+            description="Plain text of import of WiLI-2018",
+        )
+    ]
+
+    def _info(self):
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=datasets.Features(
+                {"sentence": datasets.Value("string"), "label": datasets.features.ClassLabel(names=_CLASSES)}
+            ),
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        train_path = dl_manager.download_and_extract(_TRAIN_DOWNLOAD_URL)
+        test_path = dl_manager.download_and_extract(_TEST_DOWNLOAD_URL)
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs={"filepath": train_path}),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs={"filepath": test_path}),
+        ]
+
+    def _generate_examples(self, filepath):
+
+        with open(filepath, encoding="utf-8") as f:
+            next(f)
```

Comment:
Why skip the first line here ?
The csv files have no headers


```suggestion
```

Reply:
The expected examples come to be 117499 while if I didn't add the 'next(f)' it records 117500 examples, so there was a mismatch error, so I skipped the first example.

Reply:
The number of expected samples was probably saved when you were using pandas.read_csv.
Pandas must have skipped the header and hence ignored the first example in the dataset.

Can you remove this line and regenerate the dataset_infos.json file please ?

```
datasets-cli test ./datasets/wili_2018 --save_infos --all_configs --ignore_verifications
```

Reply:
Updated that part.